src/QMCWaveFunctions/WaveFunctionPool.cpp

```diff
@@ -54,29 +54,8 @@ bool WaveFunctionPool::put(xmlNodePtr cur)
 
   ParticleSet* qp = ptcl_pool_.getParticleSet(target);
 
-  // Ye: the overall logic of the "check" is still not clear to me.
-  // The following code path should only be used when there is no cell, ion, electron info provided in the XML input.
-  // As long as any info is provided in the XML file, no need to call createESParticleSet
-  // EinsplineBuilder has code to verify cell and ion position. No need to do here.
-  if (!qp)
-  { //check ESHDF should be used to initialize both target and associated ionic system
-    xmlNodePtr tcur = cur->children;
-    while (tcur != NULL)
-    { //check <determinantset/> or <sposet_builder/> or <sposet_collection/> to extract the ionic and electronic structure
-      std::string cname((const char*)tcur->name);
-      if (cname == WaveFunctionComponentBuilder::detset_tag || cname == "sposet_builder" ||
-          cname == "sposet_collection")
-      {
-        qp = ptcl_pool_.createESParticleSet(tcur, target, qp);
-      }
-      tcur = tcur->next;
-    }
-  }
-
-  if (qp == 0)
-  {
-    APP_ABORT("WaveFunctionPool::put Target ParticleSet is not found.");
-  }
+  if(qp == nullptr)
```

Comment:
This seems like a UniformCommunicateError, why not use that std::exception subtype?  I guess then we need to figure out where is going to be caught but it looks like the ValidateXML method in QMCMain should have a `try{} catch` since this is an exception path but of a frequent sort.

Another good reason is this code will eventually be shifted to an input class and shouldn't have a dependency on Message which directly using a myComm method in causes.

Reply:
If there is a communicator available to the builder, The stop should be handled locally instead of relying on a global try somewhere.

Builder class can benefit from carrying a communicator to allow parallel construction of things if the construction needs a lot of computation which can be benefited from MPI.

Reply:
On the Second point I agree, but I don't think building and input parsing have the same necessary dependencies.  But its too big a design issue to resolve in this PR if there isn't concensus.